File: android/app/src/main/java/com/aftaab/rezumate/model/ResumeDocument.kt
package com.aftaab.rezumate.model

import kotlinx.serialization.Serializable

@Serializable
data class ResumeDocument(
    var name: String = "",
    var email: String? = null,
    var phone: String? = null,
    var website: String? = null,
    var linkedin: String? = null,
    var github: String? = null,
    var location: String? = null,
    var summary: String? = null,
    var experience: MutableList<ExperienceEntry> = mutableListOf(),
    var projects: MutableList<ProjectEntry> = mutableListOf(),
    var skillCategories: MutableList<SkillCategory> = mutableListOf(),
    var education: MutableList<EducationEntry> = mutableListOf(),
) {
    val hasContent: Boolean
        get() = name.isNotEmpty() || experience.isNotEmpty() || skillCategories.isNotEmpty()
}

@Serializable
data class ExperienceEntry(
    var company: String = "",
    var dateRange: String = "",
    var title: String = "",
    var location: String = "",
    var bullets: MutableList<String> = mutableListOf(),
)

@Serializable
data class ProjectEntry(
    var name: String = "",
    var bullets: MutableList<String> = mutableListOf(),
)

@Serializable
data class SkillCategory(
    var name: String = "",
    var items: String = "",
)

@Serializable
data class EducationEntry(
    var institution: String = "",
    var dateRange: String = "",
    var degree: String = "",
    var location: String = "",
    var details: MutableList<String> = mutableListOf(),
)

File: ios/RezumateNative/Models/ResumeDocument.swift
import Foundation

struct ResumeDocument: Equatable {
    var name: String = ""
    var email: String?
    var phone: String?
    var website: String?
    var linkedin: String?
    var github: String?
    var location: String?
    var summary: String?
    var experience: [ExperienceEntry] = []
    var projects: [ProjectEntry] = []
    var skillCategories: [SkillCategory] = []
    var education: [EducationEntry] = []
    var additionalSections: [AdditionalResumeSection] = []
    var unmappedContent: [String] = []

    var hasContent: Bool {
        !name.isEmpty || !experience.isEmpty || !skillCategories.isEmpty
    }
}

struct AdditionalResumeSection: Equatable {
    var title: String
    var lines: [String]
}

struct ExperienceEntry: Equatable {
    var company: String = ""
    var dateRange: String = ""
    var title: String = ""
    var location: String = ""
    var bullets: [String] = []
}

struct ProjectEntry: Equatable {
    var name: String = ""
    var bullets: [String] = []
}

struct SkillCategory: Equatable {
    var name: String = ""
    var items: String = ""
}

struct EducationEntry: Equatable {
    var institution: String = ""
    var dateRange: String = ""
    var degree: String = ""
    var location: String = ""
    var details: [String] = []
}
